api/server/volume.go

```diff
@@ -110,6 +111,49 @@ func (vd *volAPI) parseParam(r *http.Request, param string) (string, error) {
 	return "", fmt.Errorf("could not parse %s", param)
 }
 
+func (vd *volAPI) nodeIPtoIds(nodes []string) ([]string, error) {
+	nodeIds := make([]string, 0)
+
+	// Get cluster instance
+	c, err := clustermanager.Inst()
+	if err != nil {
+		return nodeIds, err
+	}
+
+	if c == nil {
+		return nodeIds, fmt.Errorf("failed to get cluster instance.")
+	}
+
+	for _, idIp := range nodes {
+		if idIp != "" {
+			id, err := c.GetNodeIdFromIp(idIp)
+			if err != nil {
+				return nodeIds, err
```

Comment:
nit: can be changed to 
```suggestion
				return nil, err
```

so that we either return all or none.